easybuild/easyblocks/p/pytorch.py

```diff
@@ -0,0 +1,230 @@
+##
+# Copyright 2020-2020 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing PyTorch, implemented as an easyblock
+
+@author: Alexander Grund (TU Dresden)
+"""
+
+import os
+import re
+import tempfile
+from distutils.version import LooseVersion
+from easybuild.easyblocks.generic.pythonpackage import PythonPackage
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.config import build_option
+import easybuild.tools.environment as env
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.systemtools import POWER, get_cpu_architecture
+
+
+class EB_PyTorch(PythonPackage):
+    """Support for building/installing TensorFlow."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = PythonPackage.extra_options()
+        extra_vars.update({
+            # see https://developer.nvidia.com/cuda-gpus
+            'cuda_compute_capabilities': [[], 'List of CUDA compute capabilities to build with', CUSTOM],
+            'excluded_tests': [{}, 'Mapping of architecture strings to list of tests to be excluded', CUSTOM],
+            'custom_opts': [[], 'List of options for the build/install command. Can be used to change the defaults ' +
+                                'set by the PyTorch EasyBlock, for example ["USE_MKLDNN=0"].', CUSTOM]
+        })
+        extra_vars['download_dep_fail'][0] = True
+        extra_vars['sanity_pip_check'][0] = True
+
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        super(EB_PyTorch, self).__init__(*args, **kwargs)
+        self.options['modulename'] = 'torch'
+        # Test as-if pytorch was installed
+        self.testinstall = True
+        self.tmpdir = tempfile.mkdtemp(suffix='-pytorch-build')
+
+    def fetch_step(self, skip_checksums=False):
```

Comment:
Please add a docstring: `"""Fetch sources for installing PyTorch, including those for tests."'"